mewarpx/mewarpx/utils_store/profileparser.py

```diff
@@ -0,0 +1,160 @@
+        del self.lines[0:5]
+        excl_count = self.parse_section_of_profiling_output("excl")
+        del self.lines[0:excl_count + 5]
+        incl_count = self.parse_section_of_profiling_output("incl")
+
+        self.profile_dicts.append({
+            "frame" : {"name" : "exclusive_function_profiles"},
+            "metrics" : {},
+            "children" : self.exclusive_profiles
+        })
+
+        self.profile_dicts.append({
+            "frame" : {"name" : "inclusive_function_profiles"},
+            "metrics" : {},
+            "children" : self.inclusive_profiles
+        })
+        total_count = excl_count + incl_count
+
+        print(f"Finished parsing profile output! Parsed {total_count} profiling strings.")
+
+    def parse_section_of_profiling_output(self, section):
+        '''
+        Parses either the exclusive or inclusive sections of the TinyProfiler output
+
+        Parameters
+        ----------
+
+        section (string): The section to parse the profiling output,
+            either 'excl' for exclusive or 'incl' for inclusive.
+
+        Returns
+        -------
+
+        The number of lines that were parsed
+        '''
+        if section != "excl" and section != "incl":
+            raise RuntimeError(f"'section' was not either 'excl' or 'incl', instead was {section}")
+
+        count = 0
+
+        for line in self.lines:
+            if "------" in line:
+                break
+            # these lines are of the form "Function()  ncalls  min  avg  max  max_percent"
+            # with more spaces than are here
+
+            # replace the multiple spaces with a single space
+            line = re.sub("\s+", " ", line)
+            values = line.split(" ")
+
+            # work backwards through the values in order to capture function names with spaces.
+            max_percent = float(values[-1].replace("%", ""))
+            max_time = float(values[-2])
+            avg = float(values[-3])
+            min_time = float(values[-4])
+            n_calls = int(values[-5])
+            name = "_".join(values[0:-5])
+
+            dict = {
+                "frame" : {"name" : name},
+                "metrics" : {
+                    "n_calls" : n_calls,
+                    f"{section}_min" : min_time,
+                    f"{section}_avg" : avg,
+                    f"{section}_max" : max_time,
+                    "max_percent" : max_percent
+                }
+            }
+
+            if section == "excl":
+                self.exclusive_profiles.append(dict)
+            elif section == "incl":
+                self.inclusive_profiles.append(dict)
+
+            count = count + 1
+
+        return count
+
+    def write_file(self):
+        if not self.profile_dicts:
+            raise RuntimeError("write_file called before profile data was parsed!")
+
+        print("Writing profile data json file...")
+        with open(self.write_dir + "profile_data.json", "w") as data_file:
```

Comment:
Like here for example, using `os.path.join(self.write_dir, "profile_data.json")` avoids you having to worry about whether the write directory ended with a `/`.